.test-infra/jenkins/LoadTestConfig.groovy

```diff
@@ -0,0 +1,436 @@
+
+        return newConfig
+    }
+
+    @Override
+    Map<String, Serializable> toPrimitiveValues() {
+        final def map = propertiesMap
+        verifyProperties(map)
+        return ConfigHelper.convertProperties(map)
+    }
+
+    LinkedHashMap<String, Object> getPropertiesMap() {
+        return [
+                _title: _title,
+                _test: _test,
+                _runner: _runner,
+                _pipelineOptions: _pipelineOptions
+        ]
+    }
+
+    private static void verifyProperties(final LinkedHashMap<String, Object> map) {
+        for (entry in map.entrySet()) {
+            requireNonNull(entry.value, "Missing ${entry.key.substring(1)} in configuration")
+        }
+    }
+
+    private static class PipelineOptions implements SerializableOption<Map<String, Serializable>> {
+        private String _jobName
+        private String _project
+        private String  _publishToBigQuery
+        private String  _metricsDataset
+        private String  _metricsTable
+        private InputOptions  _inputOptions
+        private def  _iterations
+        private def  _fanout
+
+        //internal usage
+        private SDK i_sdk
+        private Runner i_runner
+        private static final i_dataflowRequired = ["_numWorkers", "_tempLocation", "_autoscalingAlgorithm"]
+        private static final i_flinkRequired = ["_jobEndpoint", "_environmentType", "_environmentConfig", "_parallelism"]
+
+        //dataflow only
+        private def  _numWorkers
+        private String  _tempLocation
+        private String  _autoscalingAlgorithm
+
+        //flink only
+        private String _jobEndpoint
+        private String _environmentType
+        private String _environmentConfig
+        private def _parallelism
+
+        void jobName(final String name) { _jobName = name }
+        void project(final String project) { _project = project }
+        void tempLocation(final String location) { _tempLocation = location }
+        void publishToBigQuery(final boolean publish) { _publishToBigQuery = publish }
+        void metricsDataset(final String dataset) { _metricsDataset = dataset }
+        void metricsTable(final String table) { _metricsTable = table }
+        void inputOptions(final InputOptions options) { _inputOptions = options }
+        void iterations(final int itNumber) { _iterations = itNumber }
+        void fanout(final int fanout) { _fanout = fanout }
+        void numWorkers(final int workers) { _numWorkers = workers }
+        void autoscalingAlgorithm(final String algorithm) { _autoscalingAlgorithm = algorithm }
+        void jobEndpoint(final String endpoint) { _jobEndpoint = endpoint }
+        void environmentType(final String type) { _environmentType = type }
+        void environmentConfig(final String config) { _environmentConfig = config }
+        void parallelism(final int parallelism) { _parallelism = parallelism }
```

Comment:
There is also one more important pipeline option which is consumed by every test (both in Java and Python): `streaming`. Let's set its default value to False, so we don't have to manually specify it in all batch tests.